pkg/controller/job/indexed_job_utils.go

```diff
@@ -148,20 +148,19 @@ func (oi orderedIntervals) has(ix int) bool {
 	return oi[hi].First <= ix
 }
 
-func succeededIndexesFromJob(job *batch.Job) orderedIntervals {
-	if job.Status.CompletedIndexes == "" {
+func succeededIndexesFromJob(completedIndexes string, completions int) orderedIntervals {
```

Comment:
```suggestion
func succeededIndexesFromString(completedIndexes string, completions int) orderedIntervals {
```

or just `succeededIndexes`, but this could be a bit ambiguous?

Reply:
I opted for `succeededIndexesFromString` since `succeededIndexes` is used in a variable name in various places already, plus the function name is a bit more specific as you mentioned.